Lib/dbm/__init__.py

```diff
@@ -38,7 +38,7 @@
 class error(Exception):
     pass
 
-_names = ['dbm.gnu', 'dbm.ndbm', 'dbm.dumb']
+_names = ['dbm.sqlite3', 'dbm.gnu', 'dbm.ndbm', 'dbm.dumb']
```

Comment:
It makes dbm.sqlite3 the most preferable backend. Is it always better than dbm.gnu and dbm.ndbm?

Reply:
Yes, this is also one of Raymond's selling points in the OP (as I read him). I definitely think it should be the preferred backend.

Reply:
I think that Raymond said that it will be better than dbm.dumb, not other specialized engines.

Reply:
IMHO Sqlite is cross-platform (and Python has a many users with Windows), although likely not as fast as dbm.gnu or dbm.ndbm, since they are specialized engines especially for this use case.

So what does "better" mean? If better compatibility then Sqlite should be the first. If better performance, then it can be just before dbm.dumb.

For me both are acceptable, these are just my 50 cents.

Reply:
> I think that Raymond said that it will be better than dbm.dumb, not other specialized engines.

This is the wording from the OP; I let it speak for itself:

> Right now we support ndbm and gnu.dbm which might or might not be part of a given build. The fallback is the super slow dumbdbm. Not the sqlite3 is part of the standard build, we can do better.

Ping Raymond if you are unsure.

Reply:
Hmm basicially dbm.gnu and dbm.ndbm is key/value storage product itself but sqlite3 is more rich product than them, I am also not sure that making sqlite3 to become a most preferrable engine is right choice at this moment.

Reply:
In terms of user experience, dbm.sqlite3 is the clear winner IMO. It is available for everyone; a user can create a database using `dbm.open()` on a Linux machine, copy the generated database to a Windows or macOS machine, open the database using `dbm.open()`, and it just works. Let's ask on Discourse.

Reply:
IMO, even if we migrate the default engine to sqlite3, CPython 3.13 should be in the experimental stage.
People can argue about different performance issues or behavior issues that can not be expected at this moment.
After CPython 3.13, we can make the decision with more context.

Reply:
cc. @rhettinger

Reply:
If "better" means better compatibility then dbm.dumb should be the first. I do not think Raymond used it in this meaning.

Reply:
In order to unlock this PR, let us go for this:

```suggestion
_names = ['dbm.gnu', 'dbm.ndbm', 'dbm.sqlite3', 'dbm.dumb']
```

I'll create a discussion on Discourse, and we can chose to promote it in a follow-up PR.

Reply:
I created a poll: https://discuss.python.org/t/new-default-preferred-dbm-backend/44228